src/components/ui/InternalShadowRectButton/InternalShadowRectButton.tsx

```diff
@@ -80,16 +81,23 @@ const StyledInternalButton = styled(InternalButton)<
   `}
 `;
 
-const StyledButtonWrapper = styled(OakBox)`
+const StyledButtonWrapper = styled(OakBox)<
+  OakBoxProps & {
+    $hoverShadow?: OakDropShadowToken | null;
+  }
+>`
   .grey-shadow:has(+ * + .internal-button:focus-visible) {
     box-shadow: ${parseDropShadow("drop-shadow-centered-grey")};
   }
   .yellow-shadow:has(+ .internal-button:focus-visible) {
     box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")};
   }
-  .yellow-shadow:has(+ .internal-button:hover) {
-    box-shadow: ${parseDropShadow("drop-shadow-lemon")};
-  }
+  ${(props) => css`
+    .yellow-shadow:has(+ .internal-button:hover) {
+      box-shadow: ${parseDropShadow(props.$hoverShadow)};
+    }
+  `};
```

Comment:
🏄🏼 minor but the value is the only bit that needs interpolating here.

```suggestion
  .yellow-shadow:has(+ .internal-button:hover) {
    box-shadow: ${(props) => parseDropShadow(props.$hoverShadow)};
  }
```

Reply:
updated